### ISSUE TITLE
Is there a system to manage the N2N network?

### ISSUE THREAD

**Issue by jonahzheng** [ntop/n2n#281]
I want to monitor the status of all connections edge.

**Comment by Logan007** [comment]
The edges as well as the supernode provide basic information via their respective localhost udp management port. They default to 5644 and 5645.

You can request that information by just sending a new line (press enter) under running the following command (localhost only):

`netcat -u  localhost 5644` (edge) or
`netcat -u  localhost 5645`(supernode).

The edge even understands some simple commands, try `help` instead of just enter. 

Even though the supernode lists the currently supported and/or active communities, it only lists the total number of edges. I like the idea of listing the connected edges by their MACs under the communities. It would require an additional loop inside [this](https://github.com/ntop/n2n/blob/dev/src/sn.c#L346) loop. Are you able to add that code and want to provide a pull request?

**Comment by jonahzheng** [comment]
thank you, I'll try

**Comment by Logan007** [comment]
@jonahzheng I did not see your reply... I already did it. Not beautiful but parsable.

**Comment by tlsalex** [comment]
what is the package name for netcat , I run :
```
root@xxx yum.repos.d]# yum provides */netcat
Loaded plugins: langpacks, versionlock
Excluding 1 update due to versionlock (use "yum versionlock status" to show it)
RackTables-0.20.14-1.el7.noarch : A data-center asset management system
Repo        : epel
Matched from:
Filename    : /usr/share/RackTables/gateways/netcat
```
Should not this RackTables

**Comment by tlsalex** [comment]
I guess it is this package: nmap-ncat
after the installation, I issue:

```
[root@bdc ~]# ss -uln
State      Recv-Q Send-Q                                 Local Address:Port                                                Peer Address:Port
UNCONN     0      0                                                  *:49152                                                          *:*
UNCONN     0      0                                                  *:49152                                                          *:*
UNCONN     0      0                                          127.0.0.1:5645                                                           *:*
UNCONN     0      0                                          127.0.0.1:5645                                                           *:*
UNCONN     0      0                                          127.0.0.1:2099                                                           *:*
UNCONN     0      0                                                  *:68                                                             *:*
UNCONN     0      0                                          127.0.0.1:323                                                            *:*
UNCONN     0      0                                              [::1]:323                                                         [::]:*
[root@bdc ~]# ncat -u localhost 5645
```
The cursor get stuck there.
it returns me nothing.
Did I do something wrong ?

**Comment by Logan007** [comment]
Just send a new line by pressing the Enter key.

_P. S._ As for the package, I do not know about your distro, but you could try to get it from [here](http://netcat.sourceforge.net/). Arch Linux would be contented by `pacman -S gnu-netcat` with  `gnu-netcat` being the package name.

**Comment by tlsalex** [comment]
Now I run n2n v.2.6.0.r250 on Ubuntu 18.04.4.
It ships with netcat , I want to check the edge info from supernode , so I did bellow

```
root@bdc:~# ss -uln
State              Recv-Q              Send-Q                                Local Address:Port                              Peer Address:Port
UNCONN             0                   0                                           0.0.0.0:49152                                  0.0.0.0:*
UNCONN             0                   0                                         127.0.0.1:5645                                   0.0.0.0:*
UNCONN             0                   0                                     127.0.0.53%lo:53                                     0.0.0.0:*
UNCONN             0                   0                                  172.16.16.4%eth0:68                                     0.0.0.0:*
root@bdc:~# netcat -u localhost 5645
```
I did send a new line by press the Enter key , and tried many time ,but it returns an empty line ,
But it works well on edge by using 
`netcat -u localhost 5644`
```
Statistics for edge
uptime 1674
paths  super:6,9 p2p:950,974
transop |   956|   983|
peers  pend:0 full:1
last super:1595049193(12 sec ago) p2p:1595049205(0 sec ago)
```

 any advise ?

**Comment by tlsalex** [comment]
disable ipv6 then everything works well

**Comment by Logan007** [comment]
Good to hear. Maybe you could use `netcat -u 127.0.0.1 5644` instead…

**Comment by switch-st** [comment]
You also can check status of connetions by "kill -SIGHUP pid of supernode" on linux.

**Comment by switch-st** [comment]
@Logan007 Please take notice of this pull request. #320 

Maybe there is a better way to solve this problem.

**Comment by Logan007** [comment]
Of course, I nearly forgot about the signal…

In my case, `sudo kill -SIGHUP $(pidof supernode)` works fine. Only issue is that I have to read the stats from the log using `sudo systemctl status supernode` (or whatever your service name might be). That might be a bit hard to get by and then to parse (for further display on a web page such as seen in #17?).

I assume the issue addressed in #320 is legit and should really tried to be fixed as it might impact availability of service __or__, if neither possible nor feasible, scratch outputting communities and nodes from the supernode reverting back to the former kind of fixed length output.
